compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/tree/nodeparser/ParseIntermediateClauseTest.java

```diff
@@ -0,0 +1,333 @@
+                .parseIntermediateClause("join var user in getTable() on loginUserId equals userId", true);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.JOIN_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidOrderClause() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("order by name", false);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.ORDER_BY_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidOrderClause2() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("order by employee.name", true);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.ORDER_BY_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidOrderClause3() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("order by " +
+                "getEmployee().id", false);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.ORDER_BY_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidOrderClause4() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("order by name " +
+                "ascending, emp.id ascending", false);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.ORDER_BY_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidOrderClause5() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("order by name" +
+                " ascending, emp.id descending", true);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.ORDER_BY_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidLimitClause() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("limit 1", false);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.LIMIT_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidLimitClause2() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("limit getName()", true);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.LIMIT_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
+
+    @Test
+    public void testValidLimitClause3() {
+        IntermediateClauseNode intermediateNode = NodeParser.parseIntermediateClause("limit emp.id", true);
+        Assert.assertEquals(intermediateNode.kind(), SyntaxKind.LIMIT_CLAUSE);
+        Assert.assertFalse(intermediateNode.hasDiagnostics());
+    }
```

Comment:
Let's add one valid test with from or let clause having an action. 
e.g. `from var i in a -> w2`

And as an invalid case, let's add the same test with the allowActions=false.

Reply:
Updated